app/docs/md/patterns/form-validation.md

```diff
@@ -0,0 +1,281 @@
+
+We can use the data access in a typical route handler as follows:
+
+```javascript
+// /app/api/books.mjs
+import { upsertBook, validate } from '../models/books.mjs'
+
+export async function postBooks (req) {
+  let { problems, book } = await validate.create(req)
+
+  await upsertBook(book)
+  return {
+    location: '/books'
+  }
+}
+```
+
+The handler above is now running validation which returns any problems in the data.
+But what do we do with them?
+
+## Form Problems Loop
+
+To close the loop on server-side validation we will need a way to maintain state between requests so that we can pass those problems back and forth and fix them.
+
+We use the session to send those problems back to the front end so that the user has another chance to fix their inputs.
+
+This is the process we will use for handling problems:
+1. User submits form from `/books` that POSTS back to `/books`
+2. The post handler runs `validate()` against the form values and gets a list of `problems`.
+3. Post handler adds the problems to the session along with the initial values submitted (i.e.`session: {problems, book}`).
+4. Post handler redirects back to `/books` by setting `location: '/books'` (with the above session set).
+5. After being redirected GET API pulls the problems and book values off the session and sets them on `json` so that the page can display them.
+6. HTML page uses the `state.store.problems` and `state.store.book` to restore the form where they left off with the problems highlighted.
+
+
+The example below shows the full round trip of server validation.
+The code is annotated with the steps.
+It can be difficult to follow because the problems loop will pass through this API file several times if validation fails.
+
+
+```javascript
+// /app/api/books.mjs
+import { upsertBook, validate } from '../models/books.mjs'
+
+export async function get (req) {
+  if (req.session.problems) {
+  // 5. Back at the form we pull the problems and initial values off the session
+    let { problems, book, ...session } = req.session
+    return {
+      session,
+      // 6. The HTML page can get problems and initial values off the store
+      json: { problems, book }
+    }
+  }
+
+// 1. First user gets a blank form to fill out
+//  return { }
+}
+
+export async function post(req) {
+  const session = req.session
+  // 2. Validate form inputs and return problems
+  let { problems, book } = await validate.create(req)
+  if (problems) {
+    return {
+    // 3. Problems and initial values added to session
+      session: { ...session, problems, book },
+      // Used for progressive enhancement next module
+      json: { problems, book },
+      // 4. Redirects back to the form with the above session
+      location: '/books'
+    }
+  }
+
+  // If validation is successful the problems and old values are removed from the session
+  // eslint-disable-next-line no-unused-vars
```

Comment:
Is this eslint comment intentional for the example?

Reply:
nope. I removed it.